store/sqlstore/store.go

```diff
@@ -233,7 +240,12 @@ func New(settings model.SqlSettings, metrics einterfaces.MetricsInterface) *SqlS
 	store.stores.linkMetadata.(*SqlLinkMetadataStore).createIndexesIfNotExists()
 	store.stores.group.(*SqlGroupStore).createIndexesIfNotExists()
 	store.stores.scheme.(*SqlSchemeStore).createIndexesIfNotExists()
-	store.stores.preference.(*SqlPreferenceStore).deleteUnusedFeatures()
+
+	store.asyncMigrations = append(store.asyncMigrations, store.stores.preference.(*SqlPreferenceStore).generateAsyncMigrations()...)
+
+	if store.tryToDispatchAsyncMigrations() {
```

Comment:
This pattern seems weird to me, followed by our having to clean up in `Close`. Should we just `tryToDispatchAsyncMigrations()` and have it contain the entirety of any setup / teardown?

Reply:
if we do setup + teardown here, that does not cover all the race conditions, since the teardown empties the column, and as soon as it commits the transaction the next thing that waits on that row can pick it up and start running the same migrations again(of course this time will be a no-op)

Reply:
Maybe I'm misunderstanding the flow -- if one server starts and holds onto the lock indefinitely, how will a subsequent server upgrade successfully migrate unless the whole cluster is first powered off? (It's my understanding we support online, zero downtime upgrades at present.)

Reply:
The servers do not hold any lock while performing the migrations. The lock is only obtained in order to syncronise reads after writes(essentially guarantee that everyone will see the same value that has been written in `async_migration_runner_id` key). If the value on that key is equal to the store id then that server will run the migrations. Only 1 server will actively run the migrations and the N-1 will skip them and boot up.

There is a scenario in which we have an existing cluster that one node actively runs the async migrations, if that node hasn't finished running the migrations if a new node is added will not run any new migrations, but I think this is not something that is communicated to the customers or our customers follow, since if we deploy a new node that runs a new version of the code without rebooting the rest then the cluster becomes heterogeneous.  Do our customers follow that approach for having zero downtime upgrades? My understanding was that customers in HA formation reboot the whole cluster, and subsequently wait for all pending migrations to finish during boot up.